### PR TITLE
Criteria input should be cleared after submit

### LINKED ISSUE #2345: Criteria input should be cleared after submit
https://github.com/taskany-inc/issues/assets/9020793/139f5a54-bb4c-40e8-b925-a3bfd3ad6af6

### PR DESCRIPTION
**Pull request by LamaEats**
Add reset form button
Also criteria form will reset after submit

## PR includes

- [x] Bug Fix

## Related issues

Resolve #2345 

## QA Instructions, Screenshots, Recordings
 
https://github.com/taskany-inc/issues/assets/7002692/3fde8ae2-96ad-4649-b28f-f67eabe1c05d

### CODE REVIEW

File: src/components/CriteriaForm/CriteriaForm.module.css [hunk 1]
```diff
@@ -1,5 +1,6 @@
 .SubmitButton {
     margin-left: auto;
+    margin-right: 0.25rem;
```
Comment (DenisVorop) [inline src/components/CriteriaForm/CriteriaForm.module.css:3]: var(--gap-xs)

File: src/components/CriteriaForm/CriteriaForm.tsx [hunk 2]
```diff
@@ -337,8 +340,21 @@ export const CriteriaForm = ({
...
     }, [mode, title, selected?.id]);
 
+    const resetHandler = useCallback(() => {
+        setShowWeightInput(false);
+        reset();
+        onReset();
+    }, [reset, onReset]);
+
+    const onFormSubmitGetter = useCallback(() => {
+        return handleSubmit((values) => {
```
Comment (DenisVorop) [inline src/components/CriteriaForm/CriteriaForm.tsx:350]: why are you return function?

```ts
onFormSubmit = (values) => {
            onSubmit(values);
            resetHandler();
        }

onSubmit={handleSubmit(onFormSubmit)}
```

File: src/components/CriteriaForm/CriteriaForm.tsx [hunk 3]
```diff
@@ -409,6 +426,7 @@ export const CriteriaForm = ({
                             view="primary"
                             className={s.SubmitButton}
                         />
+                        <Button type="reset" text="Reset" view="default" />
```
Comment (DenisVorop) [inline src/components/CriteriaForm/CriteriaForm.tsx:429]: I think the reset button should be placed on the left to maintain consistency

Comment (DenisVorop) [reply src/components/CriteriaForm/CriteriaForm.tsx:429]: `text="Reset"` - i18n

**Review by DenisVorop — APPROVED** [review]

**Review by awinogradov — APPROVED** [review]

**Pull request merged**